### DISCUSSION TITLE
why "egatively acknowledged before the acknowledgment timeout"?? after Can't?

### DISCUSSION THREAD

**Discussion by startjava** [apache/pulsar#18248] [General]
https://pulsar.apache.org/docs/concepts-messaging/#negative-acknowledgement

If you are going to use negative acknowledgment on a message, make sure it is negatively acknowledged before the acknowledgment timeout.

**Comment by michaeljmarshall** [comment]
> If you are going to use negative acknowledgment on a message, make sure it is negatively acknowledged before the acknowledgment timeout.

I am pretty sure you _can_ negatively acknowledge a message after the ack timeout has triggered an internal negative acknowledgement. Are you seeing something otherwise? The documentation is just trying to say that you might negatively acknowledge the message twice in that scenario, but if you nack the message before its timeout, it will only be nack'ed once.

**Comment by startjava** [comment]
@michaeljmarshall 
It means that the nack will be retransmitted twice after the ack timeout, and only once before the ack timeout?

**Reply by michaeljmarshall** [reply]
That's correct.
